LinkedList.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  LinkedList.swift
+//  Collections
+//
+//  Adapted and modified from Airspeed Velocity and Chris Eidhof. 
+//  Created by Jeremy Corren on 12/28/16.
+//
+//
+
+/// The `LinkedList`.
+public enum LinkedList<T> {
+    
+    /// Last node in list.
+    case end
+    
+    /// Node with pointer to next node.
+    indirect case node(T, next: LinkedList<T>)
+}
+
+extension LinkedList {
+    
+    /// Construct a new list with the given `value` held by a node at the front.
+    func cons(value: T) -> LinkedList {
+        return .node(value, next: self)
+    }
+}
+
+extension LinkedList {
+    
+    /// Push a node holding the given `value` onto the front of the list.
+    public mutating func push(x: T) {
+        self = self.cons(value: x)
+    }
+    
+    /// - returns: The element contained by the node at the front of the list, if the
+    /// list is not empty. Otherwise, `nil`.
+    public mutating func pop() -> T? {
+        
+        switch self {
+        case .end: return nil
+        case let .node(x, next: xs):
+            self = xs
+            return x
+        }
+    }
+}
+
+extension LinkedList: Collection {
+    
+    // MARK: - `Collection`
+    
+    /// Index after given index `i`.
+    public func index(after: Int) -> Int {
+        return after + 1
+    }
+    
+    /// Start index.
+    public var startIndex: Int { return 0 }
```

Comment:
I've started putting everything on their own line, like:

```Swift
var startIndex: Int {
    return 0
}
```

There may be some lingering versions from my previous selves, so this might be another "do as I say not as I do" scenario.